<a href="https://colab.research.google.com/github/jooriyousif/-Wicked-Free-Wi-Fi/blob/main/Wickwd_Wifi_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Import necessary libraries



In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px

In [ ]:
pip install chart_studio

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
#libraries for downloading the graphs as html files 
import chart_studio
import chart_studio.plotly as py
import chart_studio.tools as tls

# Importing the Dataset
To import the dataset, we will use the pandas library. We will read the dataset as a dataframe and call it df.

In [ ]:
#set up for downloading the graphs as html files 
username = "Joori"
api_key = "5IoJ55GUo9Ikg6jDX0hc"

In [ ]:
chart_studio.tools.set_credentials_file(username, api_key)

In [ ]:
URL = "https://raw.githubusercontent.com/jooriyousif/-Wicked-Free-Wi-Fi/main/wicked-free-wi-fi-daily-connections.csv"

In [ ]:
df =pd.read_csv(URL)

In [ ]:
df.head()

Neighborhood                    Date  Connections
0  Charlestown  05/10/2014 12:00:00 AM           18
1  Charlestown  05/11/2014 12:00:00 AM           16
2  Charlestown  05/12/2014 12:00:00 AM           30
3  Charlestown  05/13/2014 12:00:00 AM           21
4  Charlestown  05/14/2014 12:00:00 AM           20

In [ ]:
df.tail()

Neighborhood                    Date  Connections
7463  Downtown Boston  03/01/2016 12:00:00 AM          822
7464     South Boston  03/01/2016 12:00:00 AM          292
7465    Columbus Park  03/01/2016 12:00:00 AM          158
7466      Charlestown  03/01/2016 12:00:00 AM           30
7467  City Hall Truck  03/01/2016 12:00:00 AM            0

#Cleaning and Preprocessing the Data
Next, we will clean and preprocess the data. This may involve removing missing values, converting data types, and normalizing the data. For example, we may need to convert the date column into a datetime format, so that we can perform time series analysis.

In [ ]:
# Check the data types of each column
print("Data types:")
print(df.dtypes)

# Check for missing values
print("Number of null values:")
print(df.isna().sum())

# Check for duplicates
print("Number of duplicated rows:")
print(df.duplicated().sum())


Data types:
Neighborhood    object
Date            object
Connections      int64
dtype: object
Number of null values:
Neighborhood    0
Date            0
Connections     0
dtype: int64
Number of duplicated rows:
0


In [ ]:
# Convert date column to datetime format
df['Date'] = pd.to_datetime(df['Date'])

In [ ]:
# Check the data types of each column after the change
print("Data types:")
print(df.dtypes)

Data types:
Neighborhood            object
Date            datetime64[ns]
Connections              int64
dtype: object


It looks like the data types are correct, and there are no missing or duplicated values.

#Exploratory Data Analysis
Before we start modeling the data, we will perform exploratory data analysis to gain a better understanding of the dataset. This may involve visualizing the data using graphs and charts, and calculating summary statistics.

In [ ]:
df.describe() 

Connections
count  7468.000000
mean    420.614622
std     567.112148
min       0.000000
25%      62.000000
50%     215.000000
75%     508.250000
max    3521.000000

Based on this table, we can see that there are 7,468 non-missing values in the 'Connections' column. The average number of connections per day is 420.6, with a standard deviation of 567.1. The minimum number of connections per day is 0, while the maximum is 3521.

In [ ]:
# Group by date and sum the connections
df_daily = df.groupby('Date')['Connections'].sum().reset_index()

# Create line chart with Plotly Express
fig = px.line(df_daily, x='Date', y='Connections', title='Daily Connections over Time', color_discrete_sequence=['#95b85f'])


In [ ]:
# Group by date and sum the connections
df_daily = df.groupby('Date')['Connections'].sum().reset_index()
# Create line chart with Plotly Express
line_chart= px.line(df_daily, x='Date', y='Connections', title='Daily Connections over Time',  color_discrete_sequence=['#95b85f'])
line_chart.show()

py.plot(line_chart, filename = "Daily Connections over Time", auto_open=False)

'https://plotly.com/~Joori/2/'

This code should create a density heatmap that shows the connection density by neighborhood and date.


In [ ]:
# Group the data by neighborhood and date
neighborhood_date_data = df.groupby(['Neighborhood', 'Date']).sum().reset_index()

# Create a density heatmap
heatmap = px.density_heatmap(neighborhood_date_data, x='Date', y='Neighborhood', z='Connections', color_continuous_scale='Greens',
                title='Connections by Neighborhood and Date')
heatmap.show()

py.plot(heatmap, filename = "Connections by Neighborhood and Date", auto_open=False)


'https://plotly.com/~Joori/6/'

In [ ]:
# Group the data by neighborhood and calculate the total number of connections
neighborhood_data = df.groupby('Neighborhood')['Connections'].sum().reset_index()

# Create a bar chart
bar_graph = px.bar(neighborhood_data, x='Neighborhood', y='Connections', title='Total Connections by Neighborhood', color_discrete_sequence=['#95b85f'])
bar_graph.show()
py.plot(bar_graph, filename = "Total Connections by Neighborhood", auto_open=False)



'https://plotly.com/~Joori/8/'

In [ ]:
import plotly.express as px

# Create a scatter plot
fig = px.scatter(df, x='Date', y='Connections', title='Connections Over Time', color_discrete_sequence=['#95b85f'])
fig.show()